src/main/webapp/scripts/matrixController.js

```diff
@@ -1,32 +1,179 @@
+            }else if (cellWithMask){
+                object.src = "../images/alive mask.png";
+                matrix[row][column] = 2;
+            }else{
+                object.src = "../images/alive vaccine.png";
+                matrix[row][column] = 3;
+            }
+            
+            break;
+        
+        case "infectedCell":
+            if (!cellWithMask && !cellWithVaccine){
+                object.src = "../images/Infected Cell.png";
+                matrix[row][column] = 5;
+            }else if (cellWithMask && cellWithVaccine){
+                object.src = "../images/infected mask vaccine.png";
+                matrix[row][column] = 8;
+            }else if (cellWithMask){
+                object.src = "../images/infected mask.png";
+                matrix[row][column] = 6;
+            }else{
+                object.src = "../images/infected vaccine.png";
+                matrix[row][column] = 7;
+            }
+            break;
+    }
+}
+
+function updateSelectedCellType(object){
+    selectedCellType = object.id;
+    switch(selectedCellType)
+    {
+        case "deadCell":
+            object.src = "../images/dead selected.png";
+            document.getElementById("aliveCell").src = "../images/alive normal cell.png";
+            document.getElementById("infectedCell").src = "../images/Infected Cell.png";
+            break;
+        
+        case "aliveCell":
+            object.src = "../images/live selected.png";
+            document.getElementById("deadCell").src = "../images/dead cell.png";
+            document.getElementById("infectedCell").src = "../images/Infected Cell.png";
+            break;
+        
+        case "infectedCell":
+            object.src = "../images/infected selected.png";
+            document.getElementById("deadCell").src = "../images/dead cell.png";
+            document.getElementById("aliveCell").src = "../images/alive normal cell.png";
+            break;
+    }
+}
+
+function updateSpeed(object){
+    playSpeed = 1000 - ( (parseInt(object.value) - 1) * 100 );
+    document.getElementById("matrixSpeedLabel").innerText = "Speed: " + object.value.toString();
 }
 
 function updateZoom(){
     let newZoomLevel = document.getElementById("matrixZoom").value;
+    document.getElementById('matrixZoomLabel').innerText = "Zoom: " + (newZoomLevel - 2).toString();
     cellZoom = "style = \"width: "+newZoomLevel+"em; height: "+newZoomLevel+"em\"";
-    alive = "<td class = \"matrixCell\"><img src = \"../images/alive normal cell.png\" " + cellZoom+"class = \"table-image\"></td>";
-    death = "<td class = \"matrixCell\"><img src = \"../images/dead cell.png\" "+cellZoom+"class = \"table-image\"></td>";
-    updateMatrix();
+    alive = "<td class = \"matrixCell\"><img src = \"../images/alive normal cell.png\" " + cellZoom+"class = \"table-image\" ";
+    death = "<td class = \"matrixCell\"><img src = \"../images/dead cell.png\" "+cellZoom+"class = \"table-image\" ";
```

Comment:
I found these manipulations unintuitive. I think it might be clearer for others if you operated on elements rather than manipulating strings.